File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    console.log("각 키에 할당된 문자", keymap);
    
    console.log("작성 문자열", targets);
    
    const keyCountMap = {}
    
    for(const keys of keymap){
        console.log("키조합", keys);
        
        for(let i = 0; i < keys.length; i++){
            const char = keys[i];
            
            // 이미 존재하는 값이 존재한다면 값 비교 
            // 없다면 추가 
            if(keyCountMap[char] && keyCountMap[char] < i + 1){
                continue;
            }
            
            keyCountMap[char] = i + 1;
        }
    }
    
    
     return targets.map(target => {
         const sum = [...target].reduce((acc, cur) => acc + keyCountMap[cur], 0);
         
         
         return Number.isNaN(sum) ? -1 : sum;
     })
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    key_dict = {}
    
    for key in keymap:
        for i, k in enumerate(key):
            if key_dict.get(k):
                if key_dict.get(k) > i + 1:
                    key_dict[k] = i + 1 
            else:
                key_dict[k] = i + 1
    
    result = []
    
    for target in targets:
        count = 0
        for char in target:
            index = key_dict.get(char)
            if index == None:
                count = -1
                break
            else:
                count += index
                
        result.append(count)
    
    return result